uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/pom/PomEditorDefault.java

```diff
@@ -122,14 +158,139 @@ public boolean addDependencies(List<DynamicPomDependency> deps,
         return result;
     }
 
-    private String getKeyFromDep(DynamicPomDependency dep) {
+    @Override
+    public boolean removeDependency(DynamicPomDependency dep,
+                                    Path pomPath) {
+
+        boolean result = false;
+        try {
+            org.uberfire.java.nio.file.Path filePath = Paths.get(pomPath.toURI());
+            Model model = getPOMModel(filePath);
+            List<Dependency> depsFromPom = model.getDependencies();
+            if (removeDynamicDep(depsFromPom,
+                                 dep)) {
+                model.setDependencies(depsFromPom);
+                writePOMModelOnFS(filePath,
+                                  model);
+                result = true;
+            }
+        } catch (Exception ex) {
+            logger.error("Error while removing a dependency {}:{} {} {}",
+                         dep.getArtifactID(),
+                         dep.getGroupID(),
+                         ex.getMessage(),
+                         ex);
+            result = false;
+        }
+        return result;
+    }
+
+    @Override
+    public boolean removeDependencies(List<DynamicPomDependency> deps,
```

Comment:
I think we should not even consider remove dependency operation

Reply:
This is here because we have a maven goal to check if we have unused dependencies

Reply:
@porcelli cleanup deps https://github.com/kiegroup/kie-wb-common/pull/2184

Reply:
we agreed to not support remove dependencies, so I think this should be cleaned.
@ederign correcT?

Reply:
@porcelli  @ederign I'll remove the removal